### ISSUE TITLE
Models with transparent textures break software

### ISSUE THREAD

**Issue by Cerezk** [nilooy/mixamo-animation-combiner#3]
Models that contain PNG textures with _transparency_ completely break the software when they are uploaded (whether as an FBX or GLTF/GLB file).

Steps to replicate:

1. Upload an OBJ model with a transparent texture to Mixamo and download it as an FBX.

2. Now upload the same model without the texture, and download it as a second FBX file.

3. Try uploading both models to the Animation Combiner. The one without the texture will load fine, while the one with the transparent texture will result in the window going black and the software crashing with a fatal JavaScript error.

Sample texture:

![F00_002_Onepiece_01](https://user-images.githubusercontent.com/4543835/93617182-fb453f80-fa18-11ea-93d9-c215959e13ec.png)

**Comment by nilooy** [comment]
Currently, it doesn't support transparent texture as it behaves strange sometimes with transparent texture. I will add a feature to select transparent texture but can't guarantee it will work as expected. I don't recommend using the transparent texture in the app.

**Comment by Cerezk** [comment]
Would there be any way to work around this? To separate the textures, but convert the model to GLTF/GLB and embed the animations into it, and then re-attach the textures?

I'm thinking we may not be able to use transparent textures with the tool, but that shouldn't stop us from using models that use those textures, and processing them. Can we keep the transparent textures out of the process and still process the model itself, preserving the external references to the textures?

Does that make sense?

Thanks so much for developing this tool, by the way. :) It's a brilliant idea!

**Comment by nilooy** [comment]
> Would there be any way to work around this? To separate the textures, but convert the model to GLTF/GLB and embed the animations into it, and then re-attach the textures?
> 
> I'm thinking we may not be able to use transparent textures with the tool, but that shouldn't stop us from using models that use those textures, and processing them. Can we keep the transparent textures out of the process and still process the model itself, preserving the external references to the textures?
> 
> Does that make sense?
> 
> Thanks so much for developing this tool, by the way. :) It's a brilliant idea!

Can you send me a screenshot of the error you are getting or share a sample model with transparent texture that's giving the error.

**Comment by Cerezk** [comment]
Sure, can do both! :)

[vita4-fbx.zip](https://github.com/nilooy/mixamo-animation-combiner/files/5249984/vita4-fbx.zip)

![Screenshot from 2020-09-20 00-00-44](https://user-images.githubusercontent.com/4543835/93669159-49b81400-fad5-11ea-8f95-d1dad4012cfa.png)

![Screenshot from 2020-09-20 00-01-02](https://user-images.githubusercontent.com/4543835/93669160-50df2200-fad5-11ea-89ba-5a2d048d533e.png)

**Comment by nilooy** [comment]
i'm searching for solution of the problem at the moment. In the meantime, Can you try to add the model without texture and add the texture within the app?

**Comment by nilooy** [comment]
I have fixed it by removing the option to change texture from this type of models. It works but as i have tested your model, it's bit small in size i guess, So, just zoom in after adding the character & done!!

**Comment by Cerezk** [comment]
Yeah, the small size is due to the original model actually being small in size. The software it was created in works with a different scale, and when Mixamo processed the model, it didn't change the sizing in the data (as it correctly shouldn't).

Interestingly, though, Mixamo still shows the model up-close and full size. It could be because it automatically zooms the view in on the model, making it fit the screen, perhaps?

Thanks so much for looking into this and fixing it! I'll give it a shot and let you know. ;)

**Comment by Cerezk** [comment]
Okay, whoah! End result after the conversion not good. >.<'

This is what I ended up with in A-Frame:

![Screenshot from 2020-09-21 12-31-11](https://user-images.githubusercontent.com/4543835/93729160-a9faa300-fc06-11ea-894f-40af426f195a.png)

This is what it's supposed to look like:

![Screenshot from 2020-09-21 12-32-15](https://user-images.githubusercontent.com/4543835/93729173-b41ca180-fc06-11ea-97c2-c1bef5a284e9.png)

Did the presence of the transparent textures somehow corrupt the merger/conversion process? Or is this a new bug?

**Comment by nilooy** [comment]
I'm really sorry that you are having some unexpected result. i'm using three js in the model. I have tried debugging the model with the app and found a strange object and i believe the transparent texture is causing it. I will give it an another look.

**Comment by Cerezk** [comment]
Isn't A-Frame also based on Three JS? I'm a bit surprised that Three JS is having trouble with transparent textures. I don't really understand it.

If there is anything I can do to help, let me know.

**Comment by nilooy** [comment]
I'm not saying it's the three js that's having problem but they way i structured the app, have something to do with that. you can take look at the source code if you want. After making a localcopy, just look for the file called ```loadModel.js``` in the ```src/helper``` folder. We can add a property called ```transparency``` in the ```child object```, we can also look at some issues reported on threejs repo related to transparency, as example: 
https://discourse.threejs.org/t/transparent-image-as-texture-causing-black-background/969/6 
https://discourse.threejs.org/t/material-transparency-problem/3822

**Comment by Cerezk** [comment]
Regarding getting the camera to zoom in on the loaded object (no matter what their actual size is), this seems helpful:

https://discourse.threejs.org/t/camera-zoom-to-fit-object/936

A bit off tangent, but I thought I'd mention.

**Comment by Cerezk** [comment]
Aha!

https://discourse.threejs.org/t/gltfexporter-textures-lose-alpha-solved/7873

**Comment by nilooy** [comment]
That's great, i will try it in a while.

**Comment by nilooy** [comment]
> Regarding getting the camera to zoom in on the loaded object (no matter what their actual size is), this seems helpful:
> 
> https://discourse.threejs.org/t/camera-zoom-to-fit-object/936
> 
> A bit off tangent, but I thought I'd mention.

I have tried the auto adjustment but it's doesn't work quite well but i will give another try.

**Comment by Cerezk** [comment]
> I have tried the auto adjustment but it's doesn't work quite well but i will give another try.

There are a few different options there, based on your setup, so it will take a bit of tinkering to get it to work right. This is not a priority, though -- it just makes the MCAC look a bit neater.

It's more important that we get the transparency bug fixed, so we improve upon the usability of MCAC.

That said, here is some further insight by Don McCurdy into Three JS's various modes of handling transparency in GLTF materials:

https://discourse.threejs.org/t/gltfexporter-changing-alphamode/5993/2
